src/Access/MemoryAccessStorage.cpp

```diff
@@ -90,6 +90,9 @@ bool MemoryAccessStorage::insertNoLock(const UUID & id, const AccessEntityPtr &
     auto & entries_by_name = entries_by_name_and_type[static_cast<size_t>(type)];
     auto it_by_name = entries_by_name.find(name);
     bool name_collision = (it_by_name != entries_by_name.end());
+    UUID id_by_name;
+    if (name_collision)
+        id_by_name = it_by_name->second->id;
 
     if (name_collision && !replace_if_exists)
```

Comment:
```suggestion
    if (name_collision)
        id_by_name = it_by_name->second->id;

        if (!replace_if_exists)
        {
```

Reply:
I'd prefer to keep the code like this for symmetry with later code (see the condition

```
if (id_collision && !replace_if_exists)
```

below